addons/website/models/website.py

```diff
@@ -659,44 +659,64 @@ def is_public_user(self):
     @api.model
     def viewref(self, view_id, raise_if_not_found=True):
         ''' Given an xml_id or a view_id, return the corresponding view record.
+
+            :param view_id: either a string xml_id or an integer view_id
+            :param raise_if_not_found: should the method raise an error if no view found
+            :return: The view record or empty recordset
+        '''
+        return self.viewsref([view_id], raise_if_not_found=raise_if_not_found)[view_id]
+
+    @api.model
+    def viewsref(self, keys_or_ids, raise_if_not_found=True):
```

Comment:
Aren't we going too far with those methods duplicated with "multi" versions (viewsref, are_view_active, ...) ? 
It kind of bloats the API and adds boilerplate code and deeper stacks.

This means each call for a _single_ viewref will now imply an extra function call, and wrapping/unwrapping data in a dict. Not pretty :(

Is there a real, measurable benefit to this? If we only spare a few very fast queries from time to time but we add extra jumps for most of the single-valued calls, I don't think that's good.

For me a rule of thumb would be this: if we can spare _dozens_ of queries, or some _very slow_ queries, then it is a good idea to considering refactoring the code, and - maybe! - making it more complex. But there's really a trade-off to consider in code simplicity and stack depth vs optimization.

Disclaimer: I did not see any direct call to those "multi" variants from our business code, so I don't know how many real use cases we have.

/cc @antonylesuisse 

PS: it should be `viewrefs` instead of `viewsref`, because [the principal noun](https://www.grammar-monster.com/lessons/plurals_forming_compound_nouns.htm) is ref, not view ;-)

Reply:
It was a request of Al
Purpose was to be used In all controller that use a lot of options (blog is a good example, and the current case we are working on)

With actual code in master, we make 1 req by option (8) by rendering (duplicate in py and xml). So yes we remove a dozen of requests by rendering, using simple orm_cache.
In review with Al from yesterday we reduce it, calling only once in controller view.ref and provide result to view. (8 req instead of 16, and only on cold with the cache now.
But Al say that it will be even better to only do 1 request in cold if possible... what I do with this commit.
So not so bad imo... but yes an extra step in the stack, which is acceptable for me. Other solution (that I don’t like) was to have only one method that return dict if list of key provided, else record ... like old orm in v8. But I prefer to have function that return always the same type, best practice imo, and less « magic ».

As the name of commit suggest « m »  it was not yet finish, just a ‘backup’ :) it is why it is it yet used in blog controller.
And for the name viewrefs +1  :)

Thx